app/lib/filter_query_builder.rb

```diff
@@ -0,0 +1,96 @@
+class FilterQueryBuilder
+  def initialize(facets:, user_params:)
+    @facets = facets
+    @user_params = user_params
+  end
+
+  def call
+    filters.select(&:active?).reduce({}) { |query, filter|
+      query.merge(filter.key => filter.value)
+    }
+  end
+
+private
+  attr_reader :facets, :user_params
+
+  def filters
+    @filters ||= facets.select(&:filterable).map { |f| build_filter(f) }
+  end
+
+  def build_filter(facet)
```

Comment:
As discussed IRL, the data structure being passed in here is one of the "facets" from the content item, so I think `facet` is fine.